applications/services/gui/modules/number_input.c

```diff
@@ -0,0 +1,389 @@
+#include "number_input.h"
+
+#include <gui/elements.h>
+#include <furi.h>
+#include <assets_icons.h>
+
+/** NumberInput type */
+struct NumberInput {
+    View* view;
+};
+
+typedef struct {
+    const char text;
+    const uint8_t x;
+    const uint8_t y;
+} NumberInputKey;
+
+typedef struct {
+    const char* header;
+    char* text_buffer;
+    size_t text_buffer_size;
+    bool clear_default_text;
+
+    NumberInputCallback callback;
+    void* callback_context;
+
+    int8_t selected_row;
+    uint8_t selected_column;
+} NumberInputModel;
+
+static const uint8_t keyboard_origin_x = 7;
+static const uint8_t keyboard_origin_y = 31;
+static const uint8_t keyboard_row_count = 2;
+static const uint8_t enter_symbol = '\r';
+static const uint8_t backspace_symbol = '\b';
+
+static const NumberInputKey keyboard_keys_row_1[] = {
```

Comment:
Since we talking about numbers(int) you may want to provide negative sign too.
Also it will be nice if limits(MIN/MAX) and maybe int size be adjustable